File: hello.zig
const std = @import("std");

const stdin = std.fs.File.stdin();
const stdout = std.fs.File.stdout();

pub fn main() !void {
    var in: [4096]u8 = undefined;
    var out: [4096]u8 = undefined;

    const N = try lastInt(usize, &in);

    try print("{d}\n", &out, .{N});
}

//==================================================================
// HELPERS
//==================================================================

fn lastFloat(comptime T: type, buff: []u8) !T {
    const input = try nextLine(buff);
    return try std.fmt.parseFloat(T, input, 10);
}

fn lastInt(comptime T: type, buff: []u8) !T {
    const input = try nextLine(buff);
    return try std.fmt.parseInt(T, input, 10);
}

fn next(buff: []u8) ![]u8 {
    var r = stdin.readerStreaming(buff);
    return try r.interface.takeDelimiterExclusive(' ');
}

fn nextFloat(comptime T: type, buff: []u8) !T {
    const input = try next(buff);
    return try std.fmt.parseFloat(T, input, 10);
}

fn nextInt(comptime T: type, buff: []u8) !T {
    const input = try next(buff);
    return try std.fmt.parseInt(T, input, 10);
}

fn nextLine(buff: []u8) ![]u8 {
    var r = stdin.readerStreaming(buff);
    return try r.interface.takeDelimiterExclusive('\n');
}

fn nextList(comptime T: type, buff: []u8, arr: []T, n: usize) ![]T {
    var r = stdin.readerStreaming(buff);
    for (0..(n - 1)) |i| {
        const input = try r.interface.takeDelimiterExclusive(' ');
        arr[i] = try std.fmt.parseInt(T, input, 10);
    }
    const input = try r.interface.takeDelimiterExclusive('\n');
    arr[n - 1] = try std.fmt.parseInt(T, input, 10);
    return arr;
}

fn Tuple(comptime T: type) type {
    return struct { T, T };
}

fn nextTuple(comptime T: type, buff: []u8) !Tuple(T) {
    var r = stdin.readerStreaming(buff);
    const left = try r.interface.takeDelimiterExclusive(' ');
    const right = try r.interface.takeDelimiterExclusive('\n');
    return .{
        try std.fmt.parseInt(T, left, 10),
        try std.fmt.parseInt(T, right, 10),
    };
}

fn print(comptime fmt: []const u8, buff: []u8, args: anytype) !void {
    var w = stdout.writerStreaming(buff);
    try w.interface.print(fmt, args);
    try w.interface.flush();
}

File: hello.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	n := lastInt(reader)
	fmt.Println(n)
}

//==================================================================
// HELPERS
//==================================================================

func lastFloat(reader *bufio.Reader) float64 {
	num, err := strconv.ParseFloat(nextLine(reader), 64)
	if err != nil {
		log.Fatal(err)
	}
	return num
}

func lastInt(reader *bufio.Reader) int {
	num, err := strconv.Atoi(nextLine(reader))
	if err != nil {
		log.Fatal(err)
	}
	return num
}

func next(reader *bufio.Reader) string {
	token, err := reader.ReadString(' ')
	if err != nil {
		log.Fatal(err)
	}
	return token[:len(token)-1]
}

func nextFloat(reader *bufio.Reader) float64 {
	num, err := strconv.ParseFloat(next(reader), 64)
	if err != nil {
		log.Fatal(err)
	}
	return num
}

func nextInt(reader *bufio.Reader) int {
	num, err := strconv.Atoi(next(reader))
	if err != nil {
		log.Fatal(err)
	}
	return num
}

func nextLine(reader *bufio.Reader) string {
	line, err := reader.ReadString('\n')
	if err != nil {
		log.Fatal(err)
	}
	return line[:len(line)-1]
}

func nextList(reader *bufio.Reader, arr []int, n int) {
	for i := 0; i < n - 1; i++ {
		arr[i] = nextInt(reader)
	}
	arr[n - 1] = lastInt(reader)
}

type Tuple struct {
	Left int
	Right int
}

func nextTuple(reader *bufio.Reader) Tuple {
	return Tuple{ nextInt(reader), lastInt(reader) }
}
